### ISSUE TITLE
Notifications are displayed in the Inbox

### ISSUE THREAD

**Issue by gsteimer** [tsgrp/hpi#574]
See the screencam below.  I'm not sure what changed, but notifications used to be filtered out of the Simple Workflow Inbox (labeled Periodic Review Inbox below)

![notification-inbox](https://cloud.githubusercontent.com/assets/1377228/6129352/7c77e284-b103-11e4-94f8-17b49c12f402.gif)

When making this update, we should make sure that notification tasks do not show up in the inbox and that the task count is correct as well.

@cmlewis - let me know if you think this is going to be more than a one hour update.  If it's easy, I'd like to get this into HPI 2.3.  Otherwise, I'm fine with waiting until 2.4.

**Comment by jrubins** [comment]
Not sure why this was working previously as OC does not do any filtering of the notifications from the tasks that are returned for a user. Looked back through the SVN logs for both the `getTasks` REST endpoint and the `inbox.js` file and did not see anything that changed that would cause this to happen.

Regardless, we decided that the filtering is the responsibility of HPI and not OC since the contract of `getTasks` should remain to return all the tasks for the provided user (except Wizard tasks since there is a separate endpoint for that). So I added code to filter out tasks for a user that are notifications (determined by the `workflowDefinitionName` attribute of the task).

There is a question about the default behavior of an "inbox". Should the default behavior be to filter out notifications or is this something that is opt-in on a per project basis? @gsteimer
